src/Common/FileAndFolderCheck.php

```diff
@@ -0,0 +1,89 @@
+<?php
+namespace Robo\Common;
+
+trait FileAndFolderCheck
```

Comment:
strange that it is named `FileAndFolder` while the parameter used is `fileAndDir`. I'd rename it to `FileExistenceChecker` or `ResourceExistenceChecker`